### ISSUE TITLE
Unable to run ./main run

### ISSUE THREAD

**Issue by AmandeepT** [nanoscopic/ios_remote_provider#58]
MM-C02DR2SHMD6N controlfloor % ./main run      

Registering user routes
Registering provider routes
Registering device routes
https ListenAndServe Error listen tcp: address http://localhost:8080: too many colons in address

used :

```
{
    // This listens on all interfaces on port 8080
    listen: "http://localhost:8080"

    https: true

    auth: {
        // "builtin" uses builtin database based auth
        // "mod" additionally checks for controlfloor_auth mod based auth
        type: "mod"
    }

    video: {
        // The maximum vertical size in pixels of displayed video when viewing
        //   device video in CF
        maxHeight: 1000
    }
}
```

**Comment by nanoscopic** [comment]
The "listen" option is fed directly into Golang ListenAndServe or ListenAndServeTLS functions depending on whether "https" is set to true or false. Golang then feeds that parameter directly into net.Listen. ( https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/dial.go#L625 )

That calls resolveAddrList ( https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/dial.go#L202 )

That calls internetAddrList ( https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/ipsock.go#L240 )

That calls SplitHostPort ( https://github.com/golang/go/blob/2ebe77a2fda1ee9ff6fd9a3e08933ad1ebaea039/src/net/ipsock.go#L156 )

That splits the address by "[address]:[port]". Address is either a hostname or an ipaddress ( either ipv4 or ipv6 ).

Links and explanation given here because the golang documentation is terrible and explains none of this.

TLDR: Your "listen" config is incorrect. It should be "localhost:8080". Even more to the point, why change it from the default example of ":8080" ? That also listens on localhost.

You need to also set the https setting to what you want. If you want http only, it needs to be false. Your config also doesn't make sense in that way, you specify https as true, but are putting http:// ??

**Comment by AmandeepT** [comment]
I tried with both Http and Https and both gave me same results. I will check the links you share and will let you know how it goes. The last question , I am using only MAC at the moment to run this server and provider set-up, do I need to use Linux box as well?

**Comment by sonusainier** [comment]
@AmandeepT  no you can run on MAC as well

**Comment by nanoscopic** [comment]
Change ```listen: "http://localhost:8080"``` to ```listen: "localhost:8080"```
